modules/ismp/core/src/consensus.rs

```diff
@@ -139,6 +139,15 @@ pub trait ConsensusClient {
 		proof: Vec<u8>,
 	) -> Result<(Vec<u8>, VerifiedCommitments), Error>;
 
+	/// Process the L2 state update
+	fn verify_l2_state_update(
+		&self,
+		trusted_consensus_state: Vec<u8>,
+		consensus_state_id: ConsensusStateId,
+		state_machine_map: &mut BTreeMap<StateMachine, Vec<StateCommitmentHeight>>,
+		consensus_proof: Vec<u8>
+		)-> Result<Vec<u8>, Error>;
+
```

Comment:
So this isn't really the intended direction. Rather you should construct new implementations of the `ConsensusClient` trait for the various L2 proof types. This trait implementation will be configured with the state machine id of the ethereum chain, and accepts storage proofs of ethereum, proving new finalized L2 state commitments